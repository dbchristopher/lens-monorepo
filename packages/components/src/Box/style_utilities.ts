import { CursorProperty } from 'csstype'
import { style } from 'styled-system'

export interface CursorProps {
  cursor?: CursorProperty
}

export const cursor = style({
  prop: 'cursor',
})
