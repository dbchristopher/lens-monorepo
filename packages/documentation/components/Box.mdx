---
name: Box
menu: Components
---

import {Playground} from 'docz'
import {Box} from 'looker-lens-components'

### About Box

`<Box />` is a low level component whose primary purpose is to provide a utility for easily controlling the spacing around elements in a consistent way using the [Lens spacing values](/#!/Spacing).

##### Borders

Boxes can have borders.

<Playground>
  <Box
    width="50px"
    height="50px"
    bg="palette.purple200"
    border="1px solid black"
    borderRadius="4px"
  >
    I've got borders.
  </Box>
</Playground>
