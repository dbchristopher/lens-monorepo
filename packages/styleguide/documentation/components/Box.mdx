---
name: Box
menu: Layout
---

import { Playground } from 'docz';
import { Box, List, ListItem } from 'looker-lens-components';
import BoxSpacingRecipeTable from '../../src/BoxSpacingRecipeTable';

# Box

## About

`<Box />` is a low level component whose primary purpose is to provide a utility for easily controlling the spacing around elements in a consistent way using the [Lens spacing values](/#!/Spacing).

## Spacing with Box

The `<Box />` component allows you to control it's margin and padding properties for a chosen side.

Using spacing on a `<Box />` is like following a recipe, you select:

<List type="number">
  <ListItem>The type of spacing, margin or padding</ListItem>
  <ListItem>The side to apply it to </ListItem>
  <ListItem>The amount of space to apply</ListItem>
</List>

<BoxSpacingRecipeTable />

To get the correct spacing you choose one value from each of the above columns. For example, to get 8px of margin on the right side you would do. `<Box mr="small">` or if you wanted 40px of padding on all sides you would do `<Box p="xxlarge">`.

You can combine different spacing recipes to apply spacing to multiple sides at one time. For example if you wanted 16px of margin on the left and 20px of padding on the top you would do `<Box ml="medium" pt="large">`

### Spacing Examples

Below are a few more examples of building a spacing recipe to help clarify how to use spacing with the `<Box>` component.

<Playground>
  <Box m="large">Margin All Sides - Large</Box>
  <Box pl="xsmall">Padding left - Extra Small</Box>
  <Box pl="medium">Padding left - Medium</Box>
  <Box my="xxxlarge" ml="xlarge">
    Margin Top & Bottom - XXXLarge, Margin left - XLarge
  </Box>
  <Box pb="xxlarge" ml="xxxlarge">
    {' '}
    Padding Bottom - XXLarge, Margin left - XXXLarge
  </Box>
</Playground>

### Spacing Responsively

Any one of the spacing properties can be passed an array of spacing sizes that will then apply at different breakpoints. There are 5 [breakpoints](/#!/Breakpoints) slots that you can target. Here is an example of what that looks like.

<Playground>
  <Box
    pl={[
      'xsmall', // xsmall spacing up to the first breakpoint
      'medium', // medium spacing after first breakpoint
      'xlarge', // xlarge spacing after second breakpoint
      'xxxlarge', // xxxlarge spacing after third breakpoint
    ]}
  >
    My padding on the left changes with breakpoints
  </Box>
</Playground>

## Customizing a Box

Boxes provide a low-level customization interface which allows them to have different display properties, heights, widths, background colors, positioning, etc. Here are some examples:

### Customized Display

<Playground>
  <Box display="inline">I'm inline.</Box>
  <Box display="inline">I'm also inline.</Box>
</Playground>

### Heights and Widths

<Playground>
  <Box
    display="inline-block"
    height="100px"
    bg="palette.green200"
    p="small"
    minWidth="200px"
  >
    I'm 100px tall.
  </Box>
  <Box
    display="inline-block"
    height="50px"
    bg="palette.blue200"
    ml="small"
    p="small"
    width="100px"
  >
    I'm 50px tall.
  </Box>
</Playground>

### Background Colors

Boxes can reference colors from the theme object, pulling any color on the theme's `colors` property.

<Playground>
  <Box bg="palette.purple200">My background is purple200.</Box>
  <Box bg="semanticColors.primary.main">
    I'm the semantic color "primary.main".
  </Box>
</Playground>

### Positioning

Boxes can set positioning when needed.

<Playground>
  <Box p="small" position="relative">
    <Box position="absolute" top="0" right="0" bg="palette.red300">
      I'm absolutely positioned!
    </Box>
  </Box>
</Playground>

### Borders

Boxes can have borders.

<Playground>
  <Box
    width="80px"
    height="80px"
    bg="palette.purple200"
    border="1px solid black"
    borderRadius="4px"
  >
    I've got borders.
  </Box>
</Playground>
