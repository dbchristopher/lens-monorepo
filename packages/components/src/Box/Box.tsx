import Tag from 'clean-tag'
import {UserSelectProperty} from 'csstype'
import omit from 'lodash/omit'
import React, {FunctionComponent, Ref} from 'react'
import styled, {
  css,
  CSSObject,
  FlattenSimpleInterpolation,
  StyledComponent,
} from 'styled-components'
import {
  alignContent,
  AlignContentProps,
  alignItems,
  AlignItemsProps,
  alignSelf,
  AlignSelfProps,
  border,
  borderBottom,
  BorderBottomProps,
  borderColor,
  BorderColorProps,
  borderLeft,
  BorderLeftProps,
  BorderProps,
  borderRadius,
  BorderRadiusProps,
  borderRight,
  BorderRightProps,
  borderTop,
  BorderTopProps,
  bottom,
  BottomProps,
  boxShadow,
  BoxShadowProps,
  color,
  ColorProps,
  display,
  DisplayProps,
  flex,
  flexBasis,
  FlexBasisProps,
  flexDirection,
  FlexDirectionProps,
  FlexProps,
  flexWrap,
  FlexWrapProps,
  fontFamily,
  FontFamilyProps,
  fontSize as fontSizeStyledSystem,
  fontStyle,
  FontStyleProps,
  fontWeight as fontWeightStyledSystem,
  height,
  HeightProps,
  justifyContent,
  JustifyContentProps,
  left,
  LeftProps,
  letterSpacing,
  LetterSpacingProps,
  lineHeight as lineHeightStyledSystem,
  maxHeight,
  MaxHeightProps,
  maxWidth,
  MaxWidthProps,
  minHeight,
  MinHeightProps,
  minWidth,
  MinWidthProps,
  opacity,
  OpacityProps,
  order,
  OrderProps,
  overflow,
  OverflowProps,
  position,
  PositionProps,
  right,
  RightProps,
  space,
  textAlign,
  TextAlignProps,
  top,
  TopProps,
  verticalAlign,
  VerticalAlignProps,
  width,
  WidthProps,
  zIndex,
  ZIndexProps,
} from 'styled-system'
import {
  LensFontSizeProps,
  LensFontWeightProps,
  LensLineHeightProps,
  LensSpaceProps,
  reset,
} from 'looker-lens-design-tokens'
import {Omit} from '../types'
import {cursor, CursorProps} from './style_utilities'

export interface BoxFlexProps
  extends AlignContentProps,
    AlignItemsProps,
    FlexProps,
    FlexDirectionProps,
    FlexWrapProps,
    JustifyContentProps {}

export interface BoxFlexItemProps
  extends AlignSelfProps,
    FlexBasisProps,
    FlexProps,
    OrderProps {}

export type StyledSystemCompatibleHTMLProps<T> = Omit<
  React.HTMLProps<T>,
  | 'width'
  | 'color'
  | 'height'
  | 'is'
  | 'fontSize'
  | 'fontWeight'
  | 'size'
  | 'as'
>

export interface BoxBaseProps<T>
  extends StyledSystemCompatibleHTMLProps<T>,
    LensSpaceProps,
    LensFontSizeProps,
    LensFontWeightProps,
    LensLineHeightProps,
    BorderColorProps,
    BorderProps,
    BorderRadiusProps,
    BorderRightProps,
    BorderLeftProps,
    BorderBottomProps,
    BorderTopProps,
    BottomProps,
    BoxShadowProps,
    ColorProps,
    CursorProps,
    DisplayProps,
    FontFamilyProps,
    FontStyleProps,
    HeightProps,
    LeftProps,
    LetterSpacingProps,
    MaxHeightProps,
    MaxWidthProps,
    MinHeightProps,
    MinWidthProps,
    OpacityProps,
    OverflowProps,
    PositionProps,
    RightProps,
    TextAlignProps,
    TopProps,
    VerticalAlignProps,
    WidthProps,
    ZIndexProps {
  className?: string
  animation?: FlattenSimpleInterpolation
  is?: string | JSX.Element
  ref?: React.Ref<any>
  style?: React.CSSProperties

  /**
   * Styling for :hover pseudo class.
   *
   * [MDN Reference](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Properties_Reference)
   * @example <Box hoverStyle={{border: '1px solid black'}}/>
   */
  hoverStyle?: CSSObject
  /**
   * Styling for :focus pseudo class.
   *
   * [MDN Reference](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Properties_Reference)
   * @example <Box focusStyle={{border: '1px solid black'}}/>
   */
  focusStyle?: CSSObject
  /**
   * Styling for :active pseudo class.
   *
   * [MDN Reference](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Properties_Reference)
   * @example <Box activeStyle={{border: '1px solid black'}}/>
   */
  activeStyle?: CSSObject
  /**
   * Property to set user-select CSS property
   *
   * [MDN Reference](https://developer.mozilla.org/en-US/docs/Web/CSS/user-select)
   * @example <Box userSelect="none"/>
   */
  userSelect?: UserSelectProperty
  innerRef?: any
}

export interface BoxProps<T>
  extends BoxBaseProps<T>,
    BoxFlexProps,
    BoxFlexItemProps {}

export type BoxBasePropsWithout<T, Keys> = Omit<BoxBaseProps<T>, Keys>
export type BoxPropsWithout<T, Keys> = Omit<BoxProps<T>, Keys>

type ComponentType = FunctionComponent<BoxProps<HTMLElement>>
type StyledComponentType = StyledComponent<ComponentType, BoxProps<HTMLElement>>

const pseudoClassHover = (props: BoxProps<HTMLElement>) => {
  return (
    props.hoverStyle &&
    css`
      :hover {
        ${props.hoverStyle};
      }
    `
  )
}

const pseudoClassFocus = (props: BoxProps<HTMLElement>) => {
  return (
    props.focusStyle &&
    css`
      :focus {
        ${props.focusStyle};
      }
    `
  )
}

const pseudoClassActive = (props: BoxProps<HTMLElement>) => {
  return (
    props.activeStyle &&
    css`
      :active {
        ${props.activeStyle};
      }
    `
  )
}

const userSelectCSS = (props: BoxProps<HTMLElement>) => {
  return (
    props.userSelect &&
    css`
      user-select: ${props.userSelect};
    `
  )
}

const cursorPointerOnClick = (props: BoxProps<HTMLElement>) =>
  props.onClick &&
  !props.disabled &&
  css`
    cursor: pointer;
  `

//
// Get theme from ThemeProvider
//

const BoxFactory = React.forwardRef<StyledComponentType, BoxProps<HTMLElement>>(
  (props: BoxProps<HTMLElement>, ref: Ref<StyledComponentType>) => {
    const {
      activeStyle,
      focusStyle,
      hoverStyle,
      userSelect,
      lineHeight,
      fontWeight,
      fontSize,
      innerRef,
      ...otherProps
    } = props
    return (
      <Tag
        lineHeight={lineHeight}
        fontSize={fontSize}
        fontWeight={fontWeight}
        ref={ref || innerRef}
        {...omit(otherProps, ['animation'])}
      />
    )
  }
)

/** @component */
const Box = styled<ComponentType>(BoxFactory)`
  /**
   * Global reset applied to prevent styling on top level tags outside of Lens
   * from interfering with Lens styles.
   *
   * This **must** be first.
   */
  ${reset};

  /**
   * Rules here should provide convenience styling for Box derived components.
   * Generally anything here could be overwritten by explicit values set via
   * Box's prop values. For example a function here that sets 'cursor: pointer'
   * would be overwritten by an explicit <Box cursor='copy'/>.
   */
  ${cursorPointerOnClick};

  /**
   * Pseudo classes can be styled by passing a CSS.Properties type to the
   * corresponding pseudo class helper prop. For example: <Box
   * hoverStyle={{border: '1px solid black'}}/>
   */
  ${pseudoClassHover}
  ${pseudoClassFocus}
  ${pseudoClassActive}

  /**
   * Style Utilities that extend Box's props. Most of these come from
   * styled-system but some are Lens specific.
   *
   * These should be last to override rules with lower priority.
   */
  ${alignContent};
  ${alignItems};
  ${alignSelf};
  animation: ${(props: BoxProps<HTMLElement>) => props.animation || 'none'};
  ${border};
  ${borderRadius};
  ${borderRight};
  ${borderLeft};
  ${borderBottom};
  ${borderTop};
  ${borderColor};
  ${bottom};
  ${boxShadow};
  ${color};
  ${cursor};
  ${display};
  ${flex};
  ${flexBasis};
  ${flexDirection};
  ${flexWrap};
  ${fontFamily};
  ${fontSizeStyledSystem};
  ${fontStyle};
  ${fontWeightStyledSystem};
  ${height};
  ${justifyContent};
  ${left};
  ${letterSpacing};
  ${lineHeightStyledSystem};
  ${maxHeight};
  ${maxWidth};
  ${minHeight};
  ${minWidth};
  ${opacity};
  ${order};
  ${overflow};
  ${position};
  ${right};
  ${space};
  ${textAlign};
  ${top};
  ${userSelectCSS};
  ${verticalAlign};
  ${width};
  ${zIndex};
`

export default Box
