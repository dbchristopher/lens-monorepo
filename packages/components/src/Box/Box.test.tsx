import 'jest-styled-components'
import * as React from 'react'
import {
  createWithTheme,
  mountWithTheme,
} from '../../../test/utils/create_with_theme'
import { assertSnapshot } from '../../../test/utils/snapshot'
import { Box } from './Box'

const noop = () => {
  return
}

describe('Box', () => {
  test('Box default', () => {
    assertSnapshot(<Box mt="large">ðŸ¥‘</Box>)
  })

  test('Responsive margin top Box', () => {
    assertSnapshot(<Box mt={['large', 'medium', 'small']}>ðŸ¥‘</Box>)
  })

  test('Box with null values are removed from styling', () => {
    assertSnapshot(<Box mt={['large', null, 'medium']}>ðŸ¥‘</Box>)
  })

  test('Box with SizeNone is valid', () => {
    assertSnapshot(<Box mt="none">ðŸ¥‘</Box>)
  })

  test('Box supports background-color', () => {
    assertSnapshot(<Box bg="semanticColors.primary.main" />)
  })

  test('Box supports display', () => {
    assertSnapshot(<Box display="inline-block" />)
  })

  test('Box supports height properties', () => {
    assertSnapshot(<Box height="1vh" />)
    assertSnapshot(<Box maxHeight="1vh" />)
    assertSnapshot(<Box minHeight="1vh" />)
  })

  test('Box supports width', () => {
    assertSnapshot(<Box width="1vw" />)
    assertSnapshot(<Box maxWidth="1vw" />)
    assertSnapshot(<Box minWidth="1vw" />)
  })

  test('Box supports position, top, left, bottom, right', () => {
    assertSnapshot(<Box position="absolute" />)
    assertSnapshot(<Box top="1rem" />)
    assertSnapshot(<Box left="1rem" />)
    assertSnapshot(<Box right="1rem" />)
    assertSnapshot(<Box bottom="1rem" />)
  })

  test('Box supports responsive fontSize and lineHeight', () => {
    assertSnapshot(<Box fontSize={['small', 'large']} />)
    assertSnapshot(<Box lineHeight={['small', 'large']} />)
  })

  describe('borders', () => {
    test('Box supports borders', () => {
      assertSnapshot(<Box border="1px solid black" />)
    })

    test('supports borderRadius', () => {
      assertSnapshot(<Box borderRadius="medium" />)
    })

    test('supports borderColor', () => {
      assertSnapshot(<Box borderColor="palette.charcoal200" />)
    })
  })

  describe('fonts', () => {
    test('font helpers', () => {
      assertSnapshot(<Box fontFamily="system-ui" />)
      assertSnapshot(<Box fontSize="small" />)
      assertSnapshot(<Box fontWeight="bold" />)
      assertSnapshot(<Box letterSpacing="10" />)
      assertSnapshot(<Box lineHeight="medium" />)
      assertSnapshot(<Box textAlign="right" />)
      assertSnapshot(<Box color="palette.charcoal400" />)
    })
  })

  describe('flex', () => {
    test('supports flex properties', () => {
      assertSnapshot(
        <Box
          display="flex"
          alignContent="flex-end"
          alignItems="baseline"
          justifyContent="center"
          flexWrap="wrap"
          flexDirection="column"
        />
      )
    })

    test('supports flex item properties', () => {
      assertSnapshot(
        <Box alignSelf="center" flexBasis="1" flex="1 1" order="unset" />
      )
    })
  })

  describe('is=', () => {
    test('allows Box to render as any HTML tag', () => {
      const boxAsButton = mountWithTheme(<Box is="button" />)
      expect(boxAsButton.find('button').exists()).toBeTruthy()
    })

    test('any prop can be passed to Box', () => {
      const BoxAsInput: React.FC<{ type?: string }> = props => (
        <Box is="input" {...props} />
      )
      const boxAsCheckbox = mountWithTheme(<BoxAsInput type="checkbox" />)
      expect(boxAsCheckbox.find('input[type="checkbox"]').exists()).toBeTruthy()
    })
  })

  describe('core HTML attributes', () => {
    test('Box allows autoFocus', () => {
      assertSnapshot(<Box autoFocus>Autofocus?</Box>)
    })

    test('Box allows for HTML events', () => {
      assertSnapshot(<Box onMouseEnter={noop}>Autofocus?</Box>)
    })

    test('Box allows for ARIA attributes', () => {
      assertSnapshot(<Box aria-disabled>aria-disabled</Box>)
    })
  })

  describe('cursor', () => {
    test('Box assigns cursor: pointer when onClick is defined', () => {
      const tree = createWithTheme(<Box onClick={noop} />).toJSON()
      expect(tree).toHaveStyleRule('cursor', 'pointer')
    })

    test('Box accepts a cursor property', () => {
      const tree = createWithTheme(<Box cursor="copy" />).toJSON()
      expect(tree).toHaveStyleRule('cursor', 'copy')
    })

    test("Box's cursor property overrides the pointer setting from onClick", () => {
      const tree = createWithTheme(
        <Box onClick={noop} cursor="copy" />
      ).toJSON()
      expect(tree).toHaveStyleRule('cursor', 'copy')
    })
  })

  describe('hover, focus, active', () => {
    test('Box has hover style', () => {
      const tree = createWithTheme(
        <Box hoverStyle={{ backgroundColor: 'red' }} />
      ).toJSON()
      expect(tree).toHaveStyleRule('background-color', 'red', {
        modifier: ':hover',
      })
    })

    test('Box has focus style', () => {
      const tree = createWithTheme(
        <Box focusStyle={{ backgroundColor: 'red' }} />
      ).toJSON()
      expect(tree).toHaveStyleRule('background-color', 'red', {
        modifier: ':focus',
      })
    })

    test('Box has active style', () => {
      const tree = createWithTheme(
        <Box activeStyle={{ backgroundColor: 'red' }} />
      ).toJSON()
      expect(tree).toHaveStyleRule('background-color', 'red', {
        modifier: ':active',
      })
    })
  })

  describe('user select', () => {
    test('Box cannot be selected', () => {
      const tree = createWithTheme(<Box userSelect="none" />).toJSON()
      expect(tree).toHaveStyleRule('user-select', 'none')
    })
  })
})
