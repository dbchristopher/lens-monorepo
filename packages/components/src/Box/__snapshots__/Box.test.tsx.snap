// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box Box default 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  margin-top: 1.25rem;
}

<div
  className="c0"
>
  ðŸ¥‘
</div>
`;

exports[`Box Box supports background-color 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  background-color: #64518A;
}

<div
  className="c0"
/>
`;

exports[`Box Box supports display 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  display: inline-block;
}

<div
  className="c0"
/>
`;

exports[`Box Box supports height properties 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  height: 1vh;
}

<div
  className="c0"
/>
`;

exports[`Box Box supports height properties 2`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  max-height: 1vh;
}

<div
  className="c0"
/>
`;

exports[`Box Box supports height properties 3`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  min-height: 1vh;
}

<div
  className="c0"
/>
`;

exports[`Box Box supports position, top, left, bottom, right 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  position: absolute;
}

<div
  className="c0"
/>
`;

exports[`Box Box supports position, top, left, bottom, right 2`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  top: 1rem;
}

<div
  className="c0"
/>
`;

exports[`Box Box supports position, top, left, bottom, right 3`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  left: 1rem;
}

<div
  className="c0"
/>
`;

exports[`Box Box supports position, top, left, bottom, right 4`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  right: 1rem;
}

<div
  className="c0"
/>
`;

exports[`Box Box supports position, top, left, bottom, right 5`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  bottom: 1rem;
}

<div
  className="c0"
/>
`;

exports[`Box Box supports responsive fontSize and lineHeight 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  font-size: 0.875rem;
}

@media screen and (min-width:30rem) {
  .c0 {
    font-size: 1.125rem;
  }
}

<div
  className="c0"
/>
`;

exports[`Box Box supports responsive fontSize and lineHeight 2`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  line-height: 1.25rem;
}

@media screen and (min-width:30rem) {
  .c0 {
    line-height: 1.75rem;
  }
}

<div
  className="c0"
/>
`;

exports[`Box Box supports width 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  width: 1vw;
}

<div
  className="c0"
/>
`;

exports[`Box Box supports width 2`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  max-width: 1vw;
}

<div
  className="c0"
/>
`;

exports[`Box Box supports width 3`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  min-width: 1vw;
}

<div
  className="c0"
/>
`;

exports[`Box Box with SizeNone is valid 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  margin-top: 0rem;
}

<div
  className="c0"
>
  ðŸ¥‘
</div>
`;

exports[`Box Box with null values are removed from styling 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  margin-top: 1.25rem;
}

@media screen and (min-width:48rem) {
  .c0 {
    margin-top: 1rem;
  }
}

<div
  className="c0"
>
  ðŸ¥‘
</div>
`;

exports[`Box Responsive margin top Box 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  margin-top: 1.25rem;
}

@media screen and (min-width:30rem) {
  .c0 {
    margin-top: 1rem;
  }
}

@media screen and (min-width:48rem) {
  .c0 {
    margin-top: 0.75rem;
  }
}

<div
  className="c0"
>
  ðŸ¥‘
</div>
`;

exports[`Box borders Box supports borders 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  border: 1px solid black;
}

<div
  className="c0"
/>
`;

exports[`Box borders supports borderColor 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  border-color: #DEE1E5;
}

<div
  className="c0"
/>
`;

exports[`Box borders supports borderRadius 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  border-radius: 0.25rem;
}

<div
  className="c0"
/>
`;

exports[`Box core HTML attributes Box allows autoFocus 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
}

<div
  autoFocus={true}
  className="c0"
>
  Autofocus?
</div>
`;

exports[`Box core HTML attributes Box allows for ARIA attributes 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
}

<div
  aria-disabled={true}
  className="c0"
>
  aria-disabled
</div>
`;

exports[`Box core HTML attributes Box allows for HTML events 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
}

<div
  className="c0"
  onMouseEnter={[Function]}
>
  Autofocus?
</div>
`;

exports[`Box flex supports flex item properties 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -webkit-animation: none;
  animation: none;
  -webkit-flex: 1 1;
  -ms-flex: 1 1;
  flex: 1 1;
  -webkit-flex-basis: 1;
  -ms-flex-preferred-size: 1;
  flex-basis: 1;
  -webkit-order: unset;
  -ms-flex-order: unset;
  order: unset;
}

<div
  className="c0"
/>
`;

exports[`Box flex supports flex properties 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-align-content: flex-end;
  -ms-flex-line-pack: end;
  align-content: flex-end;
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  -webkit-animation: none;
  animation: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div
  className="c0"
/>
`;

exports[`Box fonts font helpers 1`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  font-family: system-ui;
}

<div
  className="c0"
/>
`;

exports[`Box fonts font helpers 2`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  font-size: 0.875rem;
}

<div
  className="c0"
/>
`;

exports[`Box fonts font helpers 3`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  font-weight: 700;
}

<div
  className="c0"
/>
`;

exports[`Box fonts font helpers 4`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  -webkit-letter-spacing: 10;
  -moz-letter-spacing: 10;
  -ms-letter-spacing: 10;
  letter-spacing: 10;
}

<div
  className="c0"
/>
`;

exports[`Box fonts font helpers 5`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  line-height: 1.5rem;
}

<div
  className="c0"
/>
`;

exports[`Box fonts font helpers 6`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  text-align: right;
}

<div
  className="c0"
/>
`;

exports[`Box fonts font helpers 7`] = `
.c0 {
  border: 0;
  box-sizing: border-box;
  font: inherit;
  font-size: 100%;
  margin: 0;
  padding: 0;
  vertical-align: baseline;
  -webkit-animation: none;
  animation: none;
  color: #939BA5;
}

<div
  className="c0"
/>
`;
